sources/platform/actors/publishing/monetize/pay_per_event.mdx

```diff
@@ -315,54 +315,50 @@ Try to limit the number of events. Fewer events make it easier for users to unde
 
 ### Make events produce visible results
 
-Try to make your event have tangible artifacts that users can see and understand (this might not be possible when using external APIs). Each charged event should produce something concrete in the user's dataset.
+For Actors that produce data, events should map to something concrete in the user's dataset or storage.
 
-Good examples:
+However, we acknowledge that some events don't produce tangible results (such as running AI workflows or processing external API calls). This flexibility is what makes PPE pricing powerful. It gives you the freedom to charge for special operations, complex workflows, and unique value propositions.
```

Comment:
I would add something like `You can show a status message or push a record to the dataset about the non-data action performed`